### PR TITLE
Refactor upgrade codepaths step 1

### PR DESCRIPTION
**Pull request by michaelgugino**
This commit refactors some upgrade code paths.

Touched areas are:

1) Reduces usage of 'oo_all_hosts' in various places,
especially when running upgrade_control_plane.

2) Reuses common code across the various upgrade*
playbooks.

3) Moves docker upgrade checks into container_runtime_role.

4) Combines smaller playbooks and plays to reduce file sprawl.

### CODE REVIEW

File: playbooks/common/openshift-cluster/upgrades/docker/upgrade_check.yml [hunk 1]
```diff
@@ -1,58 +1 @@
...
-  when: pkg_check.rc == 0 and docker_version is not defined
-
-- name: Flag for Docker upgrade if necessary
-  set_fact:
-    l_docker_upgrade: True
-  when: pkg_check.rc == 0 and curr_docker_version.stdout | version_compare(docker_version,'<')
-
-- name: Flag to delete all images prior to upgrade if crossing Docker 1.10 boundary
-  set_fact:
-    docker_upgrade_nuke_images: True
```
Comment (michaelgugino) [inline playbooks/common/openshift-cluster/upgrades/docker/upgrade_check.yml]: This is the only place we set nuke_images and it is dependent on the boolean below, which requires that curr_docker_version < 1.10.

**Comment by mtnbikenc** [comment]
In the absence of CI upgrade tests, I wanted to run this locally at least once for a sanity check.  I'm hoping to get that done this morning as long as there are no other deployment issues.

**Comment by sdodson** [comment]
I'm working on fixing the upgrade jobs. I'd like to get those fixed before merging this, if I don't get that sorted out today then we'll re-evaluate.
/hold

**Pull request merged**